keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -190,18 +190,20 @@ class SqlArtifactRepository(
     } == 1
   }
 
-  override fun updateArtifactMetadata(name: String, type: ArtifactType, version: String, status: ArtifactStatus?, artifactMetadata: ArtifactMetadata) {
+  override fun updateArtifactMetadata(name: String, type: ArtifactType, version: String, status: ArtifactStatus?, artifactMetadata: ArtifactMetadata){
     if (!isRegistered(name, type)) {
       throw NoSuchArtifactException(name, type)
     }
 
-    return sqlRetry.withRetry(WRITE) {
+    sqlRetry.withRetry(WRITE) {
       jooq.update(ARTIFACT_VERSIONS)
         .set(ARTIFACT_VERSIONS.BUILD_METADATA, objectMapper.writeValueAsString(artifactMetadata.buildMetadata))
         .set(ARTIFACT_VERSIONS.GIT_METADATA, objectMapper.writeValueAsString(artifactMetadata.gitMetadata))
-        .where(ARTIFACT_VERSIONS.NAME.eq(name))
-        .and(ARTIFACT_VERSIONS.TYPE.eq(type))
-        .and(ARTIFACT_VERSIONS.VERSION.eq(version))
+
+        .where(
+          ARTIFACT_VERSIONS.NAME.eq(name),
+          ARTIFACT_VERSIONS.TYPE.eq(type),
+          ARTIFACT_VERSIONS.VERSION.eq(version).or(ARTIFACT_VERSIONS.VERSION.eq("$name-$version")))
```

Comment:
It would be good to add a test to cover this change in the query.

Reply:
added one test :)